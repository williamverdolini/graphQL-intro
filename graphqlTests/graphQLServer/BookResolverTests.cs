using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace graphqlTests.graphQLServer
{
    [TestFixture]
    public class BookResolverTests : GraphQLServerBaseIntegrationTest
    {
        [Test]
        public async Task Books_verify_schema()
        {
            await BuildSchema()
                .MatchSnapshotAsync();
        }

        [Test]
        public async Task BookResolver_simple_string_request_test()
        {
            IExecutionResult result = await _requestBuilder.ExecuteRequestAsync(@"
            {
                books(first:1){
                    nodes {
                    title
                    }
                    totalCount
                }
            }");
            result.ToJson().MatchSnapshot();            
        }

        [Test]
        public async Task BookResolver_simple_query_test()
        {
            IReadOnlyQueryRequest request = FirstBookQuery();
            IExecutionResult result = await _requestBuilder.ExecuteRequestAsync(FirstBookQuery());

            result.ToJson().MatchSnapshot();
        }

        [Test]
        public async Task BookResolver_simple_query_with_specific_assertion()
        {
            IReadOnlyQueryRequest request = FirstBookQuery();
            IExecutionResult result = await _requestBuilder.ExecuteRequestAsync(FirstBookQuery());

            Assert.IsNotNull(result);
            Assert.IsNull(result.Errors);
            Assert.IsNull(result.ContextData);
            Assert.IsNull(result.Extensions);
            result.ToJson().Contains("Reconstituirea");
        }

        private static IReadOnlyQueryRequest FirstBookQuery()
        {
            return QueryRequestBuilder.New()
                    .SetQuery("{ books(first:1) { nodes { title } }}")
                    .Create();
        }
    }
}