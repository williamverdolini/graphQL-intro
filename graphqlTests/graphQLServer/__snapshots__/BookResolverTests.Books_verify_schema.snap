schema {
  query: Query
}

"The node interface is implemented by entities that have a global unique identifier."
interface Node {
  id: ID!
}

"A person who wrote or co-wrote one or more books"
type Author implements Node {
  "Author Id. It's a GUID"
  id: ID!
  "Author's firstname"
  firstName: String
  "Author's surname"
  surnName: String
  "Author's web site, if she has one"
  webSite: String
  "Author's email"
  email: String
}

type AuthorizeDirective {
  policy: String
  roles: [String!]
  apply: ApplyPolicy!
}

"A connection to a list of items."
type AuthorsConnection {
  "Information to aid in pagination."
  pageInfo: PageInfo!
  "A list of edges."
  edges: [AuthorsEdge!]
  "A flattened list of the nodes."
  nodes: [Author!]
  totalCount: Int!
}

"An edge in a connection."
type AuthorsEdge {
  "A cursor for use in pagination."
  cursor: String!
  "The item at the end of the edge."
  node: Author!
}

type Book implements Node {
  id: ID!
  authors: [Author!]!
  publisher: Publisher
  relatedBooks: [Book!]!
  title: String
  abstract: String
  editionVersion: Int
  publicationDate: DateTime
  categories: [String!]
}

"A connection to a list of items."
type BooksConnection {
  "Information to aid in pagination."
  pageInfo: PageInfo!
  "A list of edges."
  edges: [BooksEdge!]
  "A flattened list of the nodes."
  nodes: [Book!]
  totalCount: Int!
}

"An edge in a connection."
type BooksEdge {
  "A cursor for use in pagination."
  cursor: String!
  "The item at the end of the edge."
  node: Book!
}

"Information about pagination in a connection."
type PageInfo {
  "Indicates whether more edges exist following the set defined by the clients arguments."
  hasNextPage: Boolean!
  "Indicates whether more edges exist prior the set defined by the clients arguments."
  hasPreviousPage: Boolean!
  "When paginating backwards, the cursor to continue."
  startCursor: String
  "When paginating forwards, the cursor to continue."
  endCursor: String
}

type Publisher implements Node {
  id: ID!
  name: String
  address: String
}

"A connection to a list of items."
type PublishersConnection {
  "Information to aid in pagination."
  pageInfo: PageInfo!
  "A list of edges."
  edges: [PublishersEdge!]
  "A flattened list of the nodes."
  nodes: [Publisher!]
  totalCount: Int!
}

"An edge in a connection."
type PublishersEdge {
  "A cursor for use in pagination."
  cursor: String!
  "The item at the end of the edge."
  node: Publisher!
}

type Query {
  "Fetches an object given its ID."
  node("ID of the object." id: ID!): Node
  "Lookup nodes by a list of IDs."
  nodes("The list of node IDs." ids: [ID!]!): [Node]!
  authors("Returns the first _n_ elements from the list." first: Int "Returns the elements in the list that come after the specified cursor." after: String "Returns the last _n_ elements from the list." last: Int "Returns the elements in the list that come before the specified cursor." before: String order: [AuthorSortInput!] where: AuthorFilterInput): AuthorsConnection
  authorById(id: String!): Author
  publishers("Returns the first _n_ elements from the list." first: Int "Returns the elements in the list that come after the specified cursor." after: String "Returns the last _n_ elements from the list." last: Int "Returns the elements in the list that come before the specified cursor." before: String order: [PublisherSortInput!] where: PublisherFilterInput): PublishersConnection
  publisherById(id: String!): Publisher
  books("Returns the first _n_ elements from the list." first: Int "Returns the elements in the list that come after the specified cursor." after: String "Returns the last _n_ elements from the list." last: Int "Returns the elements in the list that come before the specified cursor." before: String order: [BookSortInput!] where: BookFilterInput): BooksConnection
  bookById(id: String!): Book
}

"A person who wrote or co-wrote one or more books"
input AuthorFilterInput {
  and: [AuthorFilterInput!]
  or: [AuthorFilterInput!]
  "Author Id. It's a GUID"
  id: StringOperationFilterInput
  "Author's firstname"
  firstName: StringOperationFilterInput
  "Author's surname"
  surnName: StringOperationFilterInput
  "Author's web site, if she has one"
  webSite: StringOperationFilterInput
  "Author's email"
  email: StringOperationFilterInput
}

"A person who wrote or co-wrote one or more books"
input AuthorSortInput {
  "Author Id. It's a GUID"
  id: SortEnumType
  "Author's firstname"
  firstName: SortEnumType
  "Author's surname"
  surnName: SortEnumType
  "Author's web site, if she has one"
  webSite: SortEnumType
  "Author's email"
  email: SortEnumType
}

input BookFilterInput {
  and: [BookFilterInput!]
  or: [BookFilterInput!]
  id: StringOperationFilterInput
  title: StringOperationFilterInput
  abstract: StringOperationFilterInput
  editionVersion: ComparableNullableOfInt32OperationFilterInput
  publicationDate: ComparableNullableOfDateTimeOperationFilterInput
  categories: ListStringOperationFilterInput
  authors: ListStringOperationFilterInput
  publisher: StringOperationFilterInput
  relatedBooks: ListStringOperationFilterInput
}

input BookSortInput {
  id: SortEnumType
  title: SortEnumType
  abstract: SortEnumType
  editionVersion: SortEnumType
  publicationDate: SortEnumType
  publisher: SortEnumType
}

input ComparableNullableOfDateTimeOperationFilterInput {
  eq: DateTime
  neq: DateTime
  in: [DateTime]
  nin: [DateTime]
  gt: DateTime
  ngt: DateTime
  gte: DateTime
  ngte: DateTime
  lt: DateTime
  nlt: DateTime
  lte: DateTime
  nlte: DateTime
}

input ComparableNullableOfInt32OperationFilterInput {
  eq: Int
  neq: Int
  in: [Int]
  nin: [Int]
  gt: Int
  ngt: Int
  gte: Int
  ngte: Int
  lt: Int
  nlt: Int
  lte: Int
  nlte: Int
}

input ListStringOperationFilterInput {
  all: StringOperationFilterInput
  none: StringOperationFilterInput
  some: StringOperationFilterInput
  any: Boolean
}

input PublisherFilterInput {
  and: [PublisherFilterInput!]
  or: [PublisherFilterInput!]
  id: StringOperationFilterInput
  name: StringOperationFilterInput
  address: StringOperationFilterInput
}

input PublisherSortInput {
  id: SortEnumType
  name: SortEnumType
  address: SortEnumType
}

input StringOperationFilterInput {
  and: [StringOperationFilterInput!]
  or: [StringOperationFilterInput!]
  eq: String
  neq: String
  contains: String
  ncontains: String
  in: [String]
  nin: [String]
  startsWith: String
  nstartsWith: String
  endsWith: String
  nendsWith: String
}

enum ApplyPolicy {
  BEFORE_RESOLVER
  AFTER_RESOLVER
}

enum SortEnumType {
  ASC
  DESC
}

"The `@defer` directive may be provided for fragment spreads and inline fragments to inform the executor to delay the execution of the current fragment to indicate deprioritization of the current fragment. A query with `@defer` directive will cause the request to potentially return multiple responses, where non-deferred data is delivered in the initial response and data deferred is delivered in a subsequent response. `@include` and `@skip` take precedence over `@defer`."
directive @defer("If this argument label has a value other than null, it will be passed on to the result of this defer directive. This label is intended to give client applications a way to identify to which fragment a deferred result belongs to." label: String "Deferred when true." if: Boolean) on FRAGMENT_SPREAD | INLINE_FRAGMENT

"The `@specifiedBy` directive is used within the type system definition language to provide a URL for specifying the behavior of custom scalar definitions."
directive @specifiedBy("The specifiedBy URL points to a human-readable specification. This field will only read a result for scalar types." url: String!) on SCALAR

"The `@stream` directive may be provided for a field of `List` type so that the backend can leverage technology such as asynchronous iterators to provide a partial list in the initial response, and additional list items in subsequent responses. `@include` and `@skip` take precedence over `@stream`."
directive @stream("If this argument label has a value other than null, it will be passed on to the result of this stream directive. This label is intended to give client applications a way to identify to which fragment a streamed result belongs to." label: String "The initial elements that shall be send down to the consumer." initialCount: Int! = 0 "Streamed when true." if: Boolean) on FIELD

"The `DateTime` scalar represents an ISO-8601 compliant date time type."
scalar DateTime @specifiedBy(url: "https:\/\/www.graphql-scalars.com\/date-time")
